### PR TITLE
Improve Paper component

### PR DESCRIPTION
**Pull request by apedroferreira**
Closes https://github.com/mui/mui-toolpad/issues/993

- Allow building layouts inside Paper component with the new `layout` slot type
- Force Paper to take full-width of its container to keep resizing logic predictable (as it would be in the page itself)

I made Paper full-width because the resizing breaks easily and becomes very finicky if the Paper width changes when new components are added inside. To be able to resize widths inside an inner layout with content-dependent width would add a lot of edge cases to the resizing logic and make it quite more complicated, so not sure if it's worth the effort of looking into, it might be a very difficult problem to solve or may not even have a good solution. I could try if we find it really important, but I feel like it would take significant time and effort.

https://user-images.githubusercontent.com/10789765/197014848-2538e6b6-e9c8-4859-a49d-0b95fb8de28f.mov

### CODE REVIEW

**Comment by apedroferreira** [comment]
I think this is good now, it was only pending because of a bug we've fixed in the meantime.

File: packages/toolpad-components/src/Paper.tsx [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
 }
 
 export default createComponent(Paper, {
+  layoutDirection: 'both',
   argTypes: {
     elevation: {
       typeDef: { type: 'number', minimum: 0 },
       defaultValue: 1,
     },
+    direction: {
```
Comment (Janpot) [inline packages/toolpad-components/src/Paper.tsx:27]: Shouldn't we allow users to build UI in a paper just like it is in a page? If you drag at the bottom it creates an extra row, if you drag to the side it creates an extra column?

I feel like these properties `direction` and `gap` are a step back.

Comment (bytasv) [reply packages/toolpad-components/src/Paper.tsx:27]: I agree, I'd expect the behaviour that Jan described

Comment (apedroferreira) [reply packages/toolpad-components/src/Paper.tsx:27]: I thought about that, I guess I had tried it before but gave up on it for some reason. 
Might not be super easy but I'm trying again, I think it should be possible.

Comment (apedroferreira) [reply packages/toolpad-components/src/Paper.tsx]: I've made it so we can build layouts in certain slots, in this case inside the Paper component.
For that I created a new slot type, `layout`.
Let me know what you think!

**Comment by bytasv** [comment]
Hey @apedroferreira I tested this around and I think I found a weird glitch 🤔 seems that everything is working as long as you are adding items, but I tried removing item and after that I could put anything into the paper component anymore 👇 

https://user-images.githubusercontent.com/437214/197123767-bf13b990-cab2-4fb1-a670-1425ae64279f.mov

**Comment by apedroferreira** [comment]
> Hey @apedroferreira I tested this around and I think I found a weird glitch 🤔 seems that everything is working as long as you are adding items, but I tried removing item and after that I could put anything into the paper component anymore 👇
> 
>  Screen.Recording.2022-10-21.at.09.01.25.mov

There must be a few more cases where the logic depends on something being a page but now I need to account for inner layouts too.
I'll figure it out, thanks for finding this!

**Comment by apedroferreira** [comment]
> Hey @apedroferreira I tested this around and I think I found a weird glitch

I've added a fix, seems good now

**Review by bytasv — APPROVED** [review]
Seems to be working well now 👍

**Pull request merged**